Paws4Me/Paws4Me/Display Single Pet Details/View Models/petSingleViewModel.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  petSingelViewModel.swift
+//  Paws4Me
+//
+//  Created by Kirsty-Lee Walker on 2022/03/24.
+//
+
+import Foundation
+import UIKit
+
+class SinglePetViewModel {
+
+    // MARK: - Vars/Lets
+    private var singlePet: AdoptPet?
+    private var indexSinglePet: Int = 0
+
+    // MARK: - Functions
+    func setIndexForSpecificPetName (searchText: String, filteredPetObject: AdoptPet) -> Int? {
+        let indexSinglePet =  filteredPetObject.page?.firstIndex(
+            where: { $0.name?.lowercased().starts(with: searchText) ??  false })
+        return indexSinglePet
+    }
+
+    func singlePetIndex(singleSearch: Bool, filterSearch: Bool, tableView: UITableView, indexPet: Int?) -> Int {
```

Comment:
`ViewModel` should not know what a `UITableView` is 😄

Reply:
I remember, it is gone as I did not even use it anymore as I refactored it out of my ViewModel.